### PR TITLE
Add cookbook examples

### PR DESCRIPTION
**Pull request by davidbeckingsale**
https://umpire.readthedocs.io/en/feature-um-237-create-cookbook-examples/

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
Hey @mcfadden8 - can you take a look at the ReadTheDocs build?

File: examples/cookbook/recipe_three_pool_advice.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  auto allocator = rm.getAllocator("UM");
+
+  /*
+   * Create an allocator that applied "PREFFERED_LOCATION" advice to set the
+   * GPU as the preferred location.
+   */
+  auto preferred_location_allocator =
+    rm.makeAllocator<umpire::strategy::AllocationAdvisor>(
+      "preferred_location_host", allocator, "PREFERRED_LOCATION");
```
Comment (mcfadden8) [inline examples/cookbook/recipe_three_pool_advice.cpp]: How do programmers know which string to use for advice?  Are they listed someplace besides source code?

Comment (davidbeckingsale) [reply examples/cookbook/recipe_three_pool_advice.cpp]: Good point - only in the code I think. I create a ticket to document this better.

File: docs/sphinx/cookbook/no_introspection.rst [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+If you know that you won't be using any of Umpire's introspection capabalities
+for allocations that come from a particular :class:`umpire::Allocator`, you can
+turn off the introspection and avoid the overhead of tracking the associated
+metadata.
+
+.. warning::
+    Disabling introspection means that allocations from this Allocator cannot
+    be used for operations, or size and location queries.
+
```
[the remark this answers is not in the document]
Comment (davidbeckingsale) [reply docs/sphinx/cookbook/no_introspection.rst]: They just return 0. We could have them log a warning, but our error model is throwing exceptions, not returning status codes.

**Pull request merged**